packages/shared/src/components/filters/ContentTypesFilter.tsx

```diff
@@ -1,40 +1,172 @@
+        language: value,
+      });
+
+      await gqlClient.request(UPDATE_USER_PROFILE_MUTATION, {
+        data: {
+          language: value,
+        },
+      });
+    },
+    {
+      onMutate: (value) => {
+        logEvent({
+          event_name: LogEvent.ChangeSettings,
+          target_type: TargetType.Language,
+          target_id: value,
+        });
+      },
+      onSuccess: async () => {
+        const requestKeys = [
+          ...Object.values(SharedFeedPage),
+          OtherFeedPage.Preview,
+          RequestKey.Squad,
+          RequestKey.FeedPreview,
+          RequestKey.FeedPreviewCustom,
+          RequestKey.PostKey,
+          RequestKey.Bookmarks,
+          RequestKey.ReadingHistory,
+          RequestKey.RelatedPosts,
+          RequestKey.SourceFeed,
+          RequestKey.SourceMostUpvoted,
+          RequestKey.SourceBestDiscussed,
+          RequestKey.TagFeed,
+          RequestKey.TagsMostUpvoted,
+          RequestKey.TagsBestDiscussed,
+        ];
+
+        await Promise.all(
+          requestKeys.map((requestKey) => {
+            const queryKey = [requestKey];
+
+            return queryClient.invalidateQueries({
+              queryKey,
+              exact: false,
+              type: 'all',
+            });
+          }),
+        );
+      },
+      onError: () => {
+        displayToast(labels.error.generic);
+      },
+    },
+  );
+
   return (
-    <section className="flex flex-col gap-6 px-6" aria-busy={isLoading}>
-      {videoSetting && (
-        <FilterCheckbox
-          name={videoSetting.title}
-          description={videoSetting.description}
-          checked={
-            selectedSettings[videoSetting.id] ??
-            videoSetting.defaultEnabledState
-          }
-          onToggle={() =>
-            onToggleSettings(videoSetting.id, videoSetting.defaultEnabledState)
-          }
-        >
-          {videoSetting.title}
-        </FilterCheckbox>
+    <div className="flex flex-col gap-4 px-6">
+      {postTitleLanguageFeature && (
+        <>
+          <section className="flex flex-col gap-4" aria-busy={isLoading}>
+            <div className="flex flex-col">
+              <Typography
+                tag={TypographyTag.H3}
+                color={TypographyColor.Primary}
+                type={TypographyType.Body}
+                bold
+                className="mb-1"
+              >
+                Preferred language
+              </Typography>
+              <Typography
+                color={TypographyColor.Tertiary}
+                type={TypographyType.Callout}
+              >
+                Choose your preferred language for the post titles on the feed
+              </Typography>
+            </div>
+            <LanguageDropdown
+              className={{ container: 'w-full max-w-60' }}
+              name="language"
+              defaultValue={user.language}
+              onChange={(value) => onLanguageChange(value)}
+              icon={null}
```

Comment:
```suggestion
```

Reply:
Surely we can omit this?

Reply:
By default it has the icon and here it does not. I can reverse it if you want?